create/edit account & logout

blong1996 opened the pull request:
This PR allows users to create a new account from the login activity, edit there profile (Full Name, username, and password), and also log out

blong1996 committed:
Implement Create Account (User)

- add create account button and action to login activity
- add log out button to profile fragment (no function)
- remove openChatAcitivity() call from signIn() method in login
presenter

blong1996 committed:
Align profile fragment on tabs activity

ymittal commented:
Please remove the `.idea` files from the root.

ymittal commented on android/app/src/main/java/com/google/codeu/chatme/presenter/LoginActivityPresenter.java:
```diff
@@ -92,7 +92,7 @@ public void onComplete(@NonNull Task<AuthResult> task) {
                             // saves new user to real-time database
                             addUser(currentUser.getUid(), currentUser.getDisplayName());
-                            view.openChatActivity();
+
```
I think it would a good idea to open `TabsActivity` directly once an account is created. Are you thinking of something else?

ymittal commented on android/app/src/main/java/com/google/codeu/chatme/view/login/LoginActivity.java:
```diff
@@ -118,14 +120,27 @@ public void setPasswordFieldError(int err_et_password) {
         switch (view.getId()) {
 
+
+
```
Remove extra lines

ymittal commented on android/app/src/main/java/com/google/codeu/chatme/view/login/LoginActivity.java:
```diff
@@ -118,14 +120,27 @@ public void setPasswordFieldError(int err_et_password) {
     public void onClick(View view) {
+
+        String email;
+        String password;
```
Maybe initialize these strings here itself to avoid (minimal) duplication?

blong1996 committed:
Clean up login activity file, initialize email and password before switch case

blong1996 committed:
Merge branch 'dev' into create-edit-account

blong1996 committed:
Working Logout button

- Add Profile Interactor and Presenter
- Add functionality to logout button in profile fragment
- Rename buttons and edit texts in fragment
- add strings for for profile changes

blong1996 committed:
add username on create account

- create new user’s username from their email address

blong1996 committed:
comments

blong1996 committed:
Implement Edit Current User

- Allow user to update password, username, and full name (display name)

blong1996 committed:
Begin implementation of Deleting Account

- delete button added
- delete account methods added
- delete account works, however, navigation back to the login screen
isn’t working properly

blong1996 committed:
remove .idea files

blong1996 committed:
Reformat Code & Optimize Imports

ymittal commented:
Do we need a username field anymore if Firebase does not allow login with username and password as of yet? As discussed in our previous meeting, the username field should be uneditable. If there's no other use case of the field, I would suggest that we display the user's email in a `TextView`. What do you think?

blong1996 commented:
Although the username cannot be used to authenticate a user through firebase, if we plan to implement "friends" among our users, it may be best to allow the users to search for other users in the database by their username rather than ID. Therefore, a user should be allowed access to change their username in case they do not want it to be their email

ymittal commented:
That's a good point, but right now, I believe we are not checking for username uniqueness. Two new users could have `xyz@domain1.com` and `xyz@domain2.com` as their emails, but they would be assigned the same username. And even while updating the username, we would need to check for uniqueness. I think it's a good feature to have, but maybe we can work on it on a later PR. Entering a few more characters to search by email address should not be too bad, in my opinion.

ymittal commented on android/app/src/main/java/com/google/codeu/chatme/model/User.java:
```diff
@@ -5,27 +5,65 @@
+    private long timeCreated;
+    private String photoUrl;
+    private String password;
 
```
We should not have a `password` field in the `User` model to maintain security.

blong1996 committed:
removes password from User model

blong1996 commented:
well do you want to just leave it as editable for now? then in my next PR I can check for uniqueness when creating and editing a username

ymittal commented:
I would prefer if the feature is moved to a different branch, so when it's fully implemented, we can review just that PR.

blong1996 commented:
done @ymittal

blong1996 committed:
prevent username from being changed

ymittal committed:
Refactored code

ymittal committed:
Refactored code

ymittal committed:
Merge pull request #15 from ymittal/account-patch

Edit Account Details Patch

ymittal commented:
LGTM! Good work with all the Firebase functionality.
